sdk/cosmos/azure_data_cosmos/src/models/connection_string.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use std::borrow::Cow;
+
+use azure_core::{credentials::Secret, fmt::SafeDebug, Error};
+use serde::Deserialize;
+
+/// Represents a Cosmos DB connection string.
+///
+/// The [`Debug`] implementation will not print the account key.
+#[derive(Clone, Deserialize, PartialEq, Eq, SafeDebug)]
+pub struct ConnectionString {
+    pub account_endpoint: Cow<'static, str>,
+    pub account_key: Secret,
+}
+
+impl ConnectionString {
+    pub fn from_connection_string(connection_string: &str) -> Result<Self, Error> {
+        let splat = connection_string.split(';');
+        let mut account_endpoint = None;
+        let mut account_key = None;
+        for part in splat {
+            let part = part.trim();
+            if part.is_empty() {
+                continue;
+            }
+
+            let (key, value) = part.split_once('=').ok_or(Error::new(
+                azure_core::error::ErrorKind::Other,
+                "invalid connection string",
+            ))?;
+            match key {
+                "AccountEndpoint" => account_endpoint = Some(value.to_string()),
+                "AccountKey" => account_key = Some(Secret::new(value.to_string())),
+                _ => {}
```

Comment:
For .NET we added a connection string option to disable SSL cert validation to make using the emulator easier without needing a code change between emulator and prod workloads.  We should consider supporting that here too.

Reply:
I think that's worth considering but I'll file a new issue for that. We don't currently have an explicit option to configure this right now, so it's a little tricky to add. It can be configured through the `azure_core` pipeline configuration, but figuring out how to do that in a way that composes with the user's other settings is a little tricky since it requires "replacing" the HTTP client itself.